### PR TITLE
Example parser for HTTP messages

### PR DESCRIPTION
**Pull request by brian-carroll**
Another example of parsing in Roc, using a familiar format.
One day this could maybe form the lowest level of a web server framework! 🦄 

The only compiler issues I ran into were to do with `expect` evaluation.
- It tends to crash if the two sides of the `expect` don't match. For example if you compare two lists of different lengths, it often crashes. Often it's out-of-memory panics in bumpalo.
- Other times, when I do `expect actual == expected`, and `actual` is wrong, a corrupted String is printed for `expected`, containing UTF-8 replacement characters.

I couldn't get the `digits` parser to work without crashing, so for now I am leaving all of the numbers as strings. For example, instead of the OK status being `200`, it's `"200"`, for now!

### CODE REVIEW

File: examples/parser/Parser/Core.roc [hunk 1]
```diff
@@ -142,6 +143,24 @@ apply = \funParser, valParser ->
...
+##
+## This is useful if you are using a pipeline of parsers with `apply` but
+## some parsed items are not part of the final result
+##
+## >>> const (\x -> \y -> \z -> Triple x y z)
+## >>> |> apply Parser.Str.nat
+## >>> |> skip (codeunit ',')
+## >>> |> apply Parser.Str.nat
+## >>> |> skip (codeunit ',')
+## >>> |> apply Parser.Str.nat
```
Comment (rtfeldman) [inline examples/parser/Parser/Core.roc:156]: Not a blocker for this PR, but personally I like the idea of changing these to `keep` and `skip` - so they're the same number of letters, and then the two alternative choices become more obivous!

Comment (brian-carroll) [reply examples/parser/Parser/Core.roc:156]: Right, that's a good suggestion.
I guess I just didn't change much that was already there in @Qqwy's original implementation unless it was necessary.
Existing usages of `apply` are in `Parser.Core` and `parse-movies-csv.roc`.
This seems to me like it should be its own PR. I'll open one now.

Comment (brian-carroll) [reply examples/parser/Parser/Core.roc:156]: #4872

File: examples/parser/Parser/Http.roc [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+requestUri : Parser RawStr RequestUri
+requestUri =
+    codeunitSatisfies \c -> c != ' '
+    |> oneOrMore
+    |> map strFromRaw
+
+sp = codeunit ' '
+crlf = string "\r\n"
+
+# TODO: The 'digit' and 'digits' from Parser.Str are causing repl.expect to blow up
```
Comment (folkertdev) [inline examples/parser/Parser/Http.roc:74]: do you have an example of this?

**Review by folkertdev — APPROVED** [review]

File: examples/parser/Parser/Http.roc (hunk 2 above)
Comment (brian-carroll) [reply examples/parser/Parser/Http.roc:74]: Don't have one prepared but I can try to reproduce

**Pull request merged**